cmd/diag/stages/stages.go

```diff
@@ -22,37 +32,109 @@ var Command = cli.Command{
 			Flags: []cli.Flag{
 				&flags.DebugURLFlag,
 				&flags.OutputFlag,
+				&flags.AutoUpdateFlag,
+				&flags.AutoUpdateIntervalFlag,
 			},
 		},
 	},
 	Description: ``,
 }
 
+func printStages(cliCtx *cli.Context) error {
+	err := printSyncStages(cliCtx, false)
+	if err != nil {
+		util.RenderError(err)
+	}
+
+	return nil
+}
+
 func printCurentStage(cliCtx *cli.Context) error {
+	err := printSyncStages(cliCtx, true)
+	if err != nil {
+		util.RenderError(err)
+	}
+
+	return nil
+}
+
+func printSyncStages(cliCtx *cli.Context, isCurrent bool) error {
+	autoupdate := cliCtx.Bool(flags.AutoUpdateFlag.Name)
+
+	if autoupdate {
+		interval := time.Duration(cliCtx.Int(flags.AutoUpdateIntervalFlag.Name)) * time.Millisecond
+		var wg sync.WaitGroup
+		wg.Add(1)
+		defer wg.Done()
+
+		ticker := time.NewTicker(interval)
+		go func() {
```

Comment:
consider using errgroup library which allows to return error from routine and cancel others at once

Reply:
I am using only one routine, as I understood errgroup is needed to handle multiple routines